### DISCUSSION TITLE
How to use "bash" from mingw as default shell?

### DISCUSSION THREAD

**Discussion by Warchant** [community/community#26508] [Actions]
<p>I want to use bash executable from MinGW as default shell:</p>
<pre><code>C:\Program Files\Git\mingw64\bin\bash.exe
</code></pre>
<p>When I set shell: bash, it uses shell from Git: </p>
<pre><code>C:\Program Files\Git\bin\sh.exe
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/warchant">@warchant</a> ,</p>
<p>The bash on Windows for GitHub hosted machines is provided via Git Bash (packaged with Git for Windows). To custom shell, you can set the shell value to a template string using command […options] {0} […more_options].</p>
<p>eg: </p>
<pre><code>shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
</code></pre>
<p>Please check the official <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#custom-shell" rel="nofollow noopener">doc</a> for more details.</p>
<p>And here’s a refered ticket:</p>
<p><a href="https://github.com/actions/runner/issues/240" rel="nofollow noopener">https://github.com/actions/runner/issues/240</a></p>
<p>Hope it helps!</p>
